core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -223,34 +225,28 @@ ManifestFile manifest() {
   static class ContentFileStructWithMetrics implements StructLike {
     private final StructLike fileAsStruct;
     private final MetricsUtil.ReadableMetricsStruct readableMetrics;
-    private final int expectedSize;
+    private final int position;
 
     ContentFileStructWithMetrics(
-        int expectedSize,
-        StructLike fileAsStruct,
-        MetricsUtil.ReadableMetricsStruct readableMetrics) {
+        int position, StructLike fileAsStruct, MetricsUtil.ReadableMetricsStruct readableMetrics) {
       this.fileAsStruct = fileAsStruct;
       this.readableMetrics = readableMetrics;
-      this.expectedSize = expectedSize;
+      this.position = position;
     }
 
     @Override
     public int size() {
-      return expectedSize;
+      return position;
```

Comment:
This looks a bit wrong though, should we have expectedSize (projection size) as another field than position (if you are saying the two are not always the same in the Flink case).